### ISSUE TITLE
Metadata not received 

### ISSUE THREAD

**Issue by murugancmi** [amigniter/mod_audio_stream#50]
When I sent metadata from the command I could see the Freeswitch console but I didn't receive the WebSocket msg

**Bellow is my dialplan** 

local ws_url = piopiy_stream.ws_url;
  local mix_type = 'mono';
  local extra_params = session:getVariable('cmi_cmi_extraparams') or '${uuid}';
  if(piopiy_stream.listen_mode) then
    if piopiy_stream.listen_mode == 'caller_and_callee' then
      mix_type = 'mixed';
    elseif piopiy_stream.listen_mode == 'separate_channels' then
      mix_type = 'stereo';
    end
  end

  if (session:answered() == true) then
 
  else
    session:execute('set','api_on_answer=uuid_audio_stream ${uuid} start '..ws_url..' '..mix_type..' 8k testing')
    session:execute('set','api_hangup_hook=uuid_audio_stream ${uuid} stop')
    session:execute('set','rtp_disable_hold=true')
    session:answer();
    session:execute('playback','silence_stream://-1');
  end

**Comment by murugancmi** [comment]
@amigniter any input about this issue

**Comment by digipigeon** [comment]
+1 I am also not able to receive metadata. Checked packet dump.

**Comment by amigniter** [comment]
I did a test with similar lua script and as I said earlier the metadata is always properly sent. 
```lua
local ws_url = "ws://192.168.0.19:3001"
local mix_type = "mono" -- Can be "mixed" or "stereo"
local metadata = "test-metadata"

-- Check if the session is answered
if not session:answered() then
    -- Set the audio stream to start on answer
    session:execute("set", "api_on_answer=uuid_audio_stream ${uuid} start " .. ws_url .. " " .. mix_type .. " 8k " .. metadata)
    -- Set the audio stream to stop on hangup
    session:execute("set", "api_hangup_hook=uuid_audio_stream ${uuid} stop")
    -- Prevent hold audio
    session:execute("set", "rtp_disable_hold=true")
    -- Answer the call
    session:answer()
    -- Play silence to keep the call alive
    session:execute("playback", "Silent.wav")
else
    freeswitch.consoleLog("INFO", "Call already answered\n")
end
```

dialplan:
```js
<extension name="WebSocketStreamTest">
    <condition field="destination_number" expression="^9999$">
        <action application="lua" data="stream_test.lua"/>
    </condition>
</extension>
```

As I can't reproduce the issue I am closing it for now. Please reopen it if you face the same issue again and can provide me details on how to reproduce it. I will address the issue as soon as possible.

**Comment by murugancmi** [comment]
Ok i will check

**Comment by amigniter** [comment]
Actually I have found the issue. Somehow, I don't know how, I missed to push the commit I have done months ago and it remained in my local branch. All the tests I did were actually against the refactored code which was never pushed to the main branch. While addressing this, looking at the post from another user with a screenshot I noticed the code is different. I am pushing it soon, watch for the update. Thanks.
